lib/internal/timers.js

```diff
@@ -279,11 +279,11 @@ ImmediateList.prototype.append = function(item) {
 // Removes an item from the linked list, adjusting the pointers of adjacent
 // items and the linked list's head or tail pointers as necessary
 ImmediateList.prototype.remove = function(item) {
-  if (item._idleNext !== null) {
+  if (item._idleNext) {
```

Comment:
We just want to `null` and `undefined`, right? Should this be?
```suggestion
  if (item._idleNext != null) {
```

Reply:
The reason that I didn't do it like you suggest was because now it's identical to the code `LinkedList` code, which is used for the timeout/interval path. 
https://github.com/nodejs/node/blob/master/lib/internal/linkedlist.js#L15